content/docs/iac/guides/building-extending/packages/packages.md

```diff
@@ -7,49 +7,77 @@ meta_image: /images/docs/meta-images/docs-meta.png
-3. **Component Package:** Write a Pulumi Component in your language of choice and expose it to users in all Pulumi languages. Example: the [`eks`](/registry/packages/eks) package.
+## Authoring packages
+
+There are two common cases for authoring packages:
+
+1. You are authoring a Pulumi component to be shared within your team, organization, or by anyone in the Pulumi community.
+1. You are authoring a Pulumi provider that allows your package's consumers to manage resources for a cloud or SaaS provider. (You might optionally publish this provider in the Pulumi Registry if you intend it for public consumption.)
+
+### Authoring a component for distribution
+
+If you are authoring a Pulumi component to be shared within your team or organization, you will need to decide whether to use local packages or publish SDKs. **Most component authors will want consumers to use local packages** for the following reasons:
+
+- The overhead of publishing SDKs can be significant: your CI/CD process will need to generate SDKs for all Pulumi languages (or at least all the languages your package consumers will use) and you will need package feeds to host those published SDKs.
+- Pulumi only offers tooling for writing packages with published SDKs in Go, the [Pulumi Provider SDK](/docs/iac/guides/building-extending/providers/pulumi-provider-sdk/).
+
+For an example of building and publishing a component with local packages, see [Build a Component](/docs/iac/guides/building-extending/components/build-a-component/).
+
+{{% notes type="info" %}}
+It is technically possible to author a Pulumi provider in any language, create a hand-authored schema, then generate and publish SDKs from the hand-authored schema. Most organizations will find this approach difficult to manage at scale.
+{{% /notes %}}
+
+However, there are some circumstances when using Pulumi Provider SDK and publishing SDKs may be a better option:
+
+- The requirement for consumers to install the package language's runtime is onerous. This may be because you have a large number of development teams that all want to write Pulumi in different languages, or you are writing a component for public consumption. The [Pulumi EKS package](/registry/packages/eks/) is an example of the latter use case.
+- Your team is comfortable writing and maintaining code in Go.
+- Your organization already has the package feeds necessary to host SDKs in all languages in which the component may be consumed.
+
+### Authoring a Pulumi provider
+
+If you are authoring a Pulumi provider that allows consumers to manage resources for a new cloud or SaaS provider, you should author your provider in Go using [Pulumi Provider SDK](/docs/iac/guides/building-extending/providers/pulumi-provider-sdk/).
```

Comment:
Maybe worth point out that we could support other languages, and in fact both C# and Python do have some support here, just no auto-schema generation yet, it's just a case of prioritisation so big/many customers could influence direction here.